libcrux-psq/src/handshake/ciphersuite/responder.rs

```diff
@@ -47,155 +75,165 @@ impl<'a> CiphersuiteBase for ResponderCiphersuite<'a> {
+    //     //         unreachable!("unsupported ciphersuite")
+    //     //     }
+    //     // };
+
+    //     // let ikm = K0Ikm {
+    //     //     g_xs: &DHSharedSecret::derive(own_sk, peer_pk)?,
+    //     // };
+
+    //     // Ok(AEADKey::new(&ikm, &tx0)?)
+    // }
+
+    // fn k1(&self) -> Result<AEADKey, HandshakeError> {
+    //     todo!()
+    // }
+
+    // fn tx2(&self) -> Result<Transcript, HandshakeError> {
+    //     todo!()
+    // }
+
+    // fn k2(&self) -> Result<AEADKey, HandshakeError> {
+    //     todo!()
+    // }
+
+    // fn client_authenticator(&self) -> super::types::Authenticator {
+    //     todo!()
+    // }
+}
+impl<'a> ResponderCiphersuite<'a> {
+    pub(crate) fn aead_type(&self) -> AEAD {
+        self.aead_type
     }
-    fn own_ecdh_encapsulation_key(&self) -> &DHPublicKey {
-        match self {
-            ResponderCiphersuite::X25519MlKem768ChaCha20Poly1305HkdfSha256(
-                responder_x25519_mlkem768_chacha_poly_hkdf_sha256,
-            ) => {
-                &responder_x25519_mlkem768_chacha_poly_hkdf_sha256
-                    .longterm_ecdh_keys
-                    .pk
-            }
-            #[cfg(feature = "classic-mceliece")]
-            ResponderCiphersuite::X25519ClassicMcElieceChaCha20Poly1305HkdfSha256(
-                responder_x25519_classic_mc_eliece_chacha_poly_hkdf_sha256,
-            ) => {
-                &responder_x25519_classic_mc_eliece_chacha_poly_hkdf_sha256
-                    .longterm_ecdh_keys
-                    .pk
-            }
-            #[cfg(not(feature = "classic-mceliece"))]
-            ResponderCiphersuite::X25519ClassicMcElieceChaCha20Poly1305HkdfSha256(_) => {
-                // We can never reach this because the ciphersuite can only be constructed with the feature turned on.
-                unreachable!("unsupported ciphersuite")
+    pub(crate) fn verify(
+        &self,
+        vk: &SigVerificationKey,
+        tx1: &Transcript,
+        signature: &Signature,
+    ) -> Result<(), HandshakeError> {
+        let payload = tx1.tls_serialize().map_err(HandshakeError::Serialize)?;
+        match (vk, signature) {
+            (SigVerificationKey::Ed25519(verification_key), Signature::Ed25519(sig)) => {
+                libcrux_ed25519::verify(&payload, verification_key.as_ref(), sig)
+                    .map_err(|e| e.into())
             }
-            ResponderCiphersuite::X25519NoneChaCha20Poly1305HkdfSha256(
-                responder_x25519_cha_cha_poly_hkdf_sha256,
-            ) => {
-                &responder_x25519_cha_cha_poly_hkdf_sha256
-                    .longterm_ecdh_keys
-                    .pk
+
+            (
+                SigVerificationKey::MlDsa65(mldsaverification_key),
+                Signature::MlDsa65(mldsasignature),
+            ) => libcrux_ml_dsa::ml_dsa_65::verify(
+                mldsaverification_key,
+                &payload,
+                b"PSQ",
```

Comment:
This is hard coded on both sides. If you move it out to a constant, it would be a little less error prone.